### PR TITLE
Update CHANGELOG.md to 0.10.4

### PR DESCRIPTION
**Pull request by harrison-tin**
**What this PR does / why we need it**:
Update CHANGELOG.md to 0.10.4 for upcoming release.
**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR has an associated PR with documentation in [akri-docs](https://github.com/project-akri/akri-docs)
- [ ] this PR contains unit tests
- [ ] added code adheres to standard Rust formatting (`cargo fmt`)
- [ ] code builds properly (`cargo build`)
- [ ] code is free of common mistakes (`cargo clippy`)
- [ ] all Akri tests succeed (`cargo test`)
- [ ] inline documentation builds (`cargo doc`)
- [ ] all commits pass the [DCO bot check](https://probot.github.io/apps/dco/) by being signed off -- see the failing DCO check for instructions on how to retroactively sign commits

### CODE REVIEW

**Comment by kate-goldenring** [comment]
@harrison-tin version should be bumped to 0.10.3

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,93 @@
...
+## Announcing Akri v0.10.2!
+Akri v0.10.2 is a pre-release of Akri.
+
+To find out more about Akri, check out our [documentation](https://docs.akri.sh/) and start
+[contributing](https://docs.akri.sh/community/contributing) today!
+
+## New Features
+The v0.10.2 release contains the following changes:
+
+1. **Enable mounting connectivity information for multiple devices/instances in a Pod** (https://github.com/project-akri/akri/pull/560 , https://github.com/project-akri/akri/pull/561). Previously, Akri could only mount one device property per protocol to a Pod as all devices of the same protocol had the same environment variable name. This release fixes this issue by appending the instance hash to the environment variable name and slot ID to the annotation key name. This is a **breaking change** as it changes the way brokers look up properties.
```
Comment (kate-goldenring) [inline CHANGELOG.md:12]: protocol -> "discovery handler"

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,93 @@
...
+2. **Enable udev discovery handler to discover multiple node devices** (https://github.com/project-akri/akri/pull/564). Akri now allows udev discovery handler to group devices that share a parent/child relation.
+3. **Mount udev devpath in Akri brokers** (https://github.com/project-akri/akri/pull/534). This enables discovering udev devices without a devnode by using devpath instead. This is a **breaking change** in the udev discovery handler as it changes the way Akri creates instance ids for udev devices.
+
```
Comment (kate-goldenring) [inline CHANGELOG.md:15]: We should also include this as a highlighted change: https://github.com/project-akri/akri/pull/576.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,93 @@
...
+**Fixes, features, and optimizations**
+- fix: OPC discovery handler uses discoveryURL specified by users if the return applicationURL from OPC server is not resolvable (https://github.com/project-akri/akri/pull/570)
+- fix: Fixed watch crash API unreachable (https://github.com/project-akri/akri/pull/568)
+- fix: Added udev devnode to device mounts instead of devpath (https://github.com/project-akri/akri/pull/562)
+- fix: Modify Agent to reduce frequency of Pods getting UnexpectedAdmissionError (https://github.com/project-akri/akri/pull/556)
+- opt: Use tokio::sync::RwLock instead of tokio::sync::Mutex (https://github.com/project-akri/akri/pull/541)
+- opt: ListAndWatch only sends device if the list has changed (https://github.com/project-akri/akri/pull/540)
+- opt: Added more securityContext to ensure Helm templates use the most restrictive setting (https://github.com/project-akri/akri/pull/547)
+- feat: Add nodeSelectors for Akri agent (https://github.com/project-akri/akri/pull/536)
+
```
Comment (kate-goldenring) [inline CHANGELOG.md:25]: I believe we have a few more that have recently come in. I'd look at the latest merged PRs or this changelog: https://github.com/project-akri/akri/compare/v0.8.23...main

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,93 @@
...
+
+View the [full change log](https://github.com/project-akri/akri/compare/v0.8.23...v0.10.2)
+
+## Breaking Changes
+1. With [Enable mounting connectivity information for multiple devices/instances in a Pod](https://github.com/project-akri/akri/pull/561), Akri now changes the name of the device properties from DEVICE_DESCRIPTION to DEVICE_DESCRIPTION_INSTANCE_HASH to allow multiple device properties of the same protocol to be injected to the same broker. For example, broker can look up the akri instance `akri-debug-echo-foo-8120fe` by the environment variable `DEBUG_ECHO_DESCRIPTION_8120FE` instead of `DEBUG_ECHO_DESCRIPTION`.
```
Comment (kate-goldenring) [inline CHANGELOG.md:30]: nit: "enable"

Comment (kate-goldenring) [inline CHANGELOG.md:30]: nit: capitalize Akri

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,93 @@
...
+| MicroK8s | 1.23/stable |
+
+## What's next?
+Check out our [roadmap](https://docs.akri.sh/community/roadmap) to see the features we are looking forward to!
+
+## Thanks 👏
+Thank you everyone in the community who helped Akri get to this release! Your interest and contributions help Akri
+prosper.
+
+**⭐ Contributors to v0.10.x ⭐**
```
Comment (kate-goldenring) [inline CHANGELOG.md:60]: should this be v0.10.3 or is the x intentional?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,93 @@
...
+- @adithyaj
+- @kate-goldenring
+- @johnsonshih
+- @diconico07
+- @jbpaux
+- @yujinkim-msft
+- @harrison-tin
+- @koutselakismanos
+
```
Comment (kate-goldenring) [inline CHANGELOG.md:69]: @harrison-tin i'd put your handle here since you are making the release

File: CHANGELOG.md (hunk 3 above)
Comment (harrison-tin) [reply CHANGELOG.md:33]: updated to add more fixes. also sorting this list from old to new

**Comment by harrison-tin** [comment]
> @harrison-tin version should be bumped to 0.10.3 and the change log link

@kate-goldenring bumping to 0.10.4 since we merged https://github.com/project-akri/akri/pull/585 as well

File: CHANGELOG.md
Comment (adithyaj) [inline CHANGELOG.md:1]: nit but line 304 change the last part of the changelog url to "compare/v0.7.0...**v**0.8.4"

**Review by adithyaj — APPROVED** [review]

File: CHANGELOG.md
Comment (harrison-tin) [reply CHANGELOG.md:1]: thanks for the approval. line 304 is the changelog for release v0.7.0 so v0.6.19...v0.7.0 seems correct.

**Pull request merged**